### PR TITLE
test: Cypress 7.x migration

### PR DESCRIPTION
**Pull request by lyndsiWilliams**
### SUMMARY
This PR migrates Cypress to 7.x. Details for this process can be found [here](https://docs.cypress.io/guides/references/migration-guide#Migrating-to-Cypress-7-0).

### TESTING INSTRUCTIONS
- Navigate to `superset-frontend/package.json` and observe the dev dependency: `"@cypress/react": "^5.10.0"`
- Navigate to `superset-frontend/cypress-base/package.json` and observe the dependency: `"@cypress/code-coverage": "^3.9.11"` and dev dependency: `"cypress": "^7.0.0"`
- Run the Cypress tests and observe that tests have been properly migrated.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/cypress-base/package.json [hunk 1]
```diff
@@ -10,14 +10,15 @@
...
+    "@cypress/code-coverage": "^3.9.11",
     "@superset-ui/core": "^0.17.42",
     "react-dom": "^16.13.0",
     "rison": "^0.1.1",
     "shortid": "^2.2.15"
   },
   "devDependencies": {
-    "cypress": "^6.3.0",
+    "@cypress/webpack-dev-server": "^1.6.0",
+    "cypress": "^6.9.1",
```
Comment (eschutho) [inline superset-frontend/cypress-base/package.json:21]: should this be 7x?

File: superset-frontend/cypress-base/package.json [hunk 2]
```diff
@@ -10,14 +10,15 @@
...
-    "@cypress/code-coverage": "^3.9.2",
+    "@cypress/code-coverage": "^3.9.11",
     "@superset-ui/core": "^0.17.42",
     "react-dom": "^16.13.0",
     "rison": "^0.1.1",
     "shortid": "^2.2.15"
   },
   "devDependencies": {
-    "cypress": "^6.3.0",
+    "@cypress/webpack-dev-server": "^1.6.0",
```
Comment (eschutho) [inline superset-frontend/cypress-base/package.json:20]: curious, what happens if you remove this dependency?

File: superset-frontend/cypress-base/package.json (hunk 1 above)
Comment (lyndsiWilliams) [reply superset-frontend/cypress-base/package.json]: Whoops, I did everything except the most important thing! 😅 
Fixed in [`this commit`](https://github.com/apache/superset/pull/16734/commits/5c1fc6feccfa3e0fbb1b6306be1964793780586c).

File: superset-frontend/cypress-base/package.json (hunk 2 above)
Comment (lyndsiWilliams) [reply superset-frontend/cypress-base/package.json]: I ran all the e2e tests with and without this dependency. There are actually more passing tests without this, so I've removed it.

**Comment by lyndsiWilliams** [comment]
Continued at [`this PR`](https://github.com/apache/superset/pull/16911).

**Pull request closed without merging**